### ISSUE TITLE
Check if elasticsearch is adding documents

### ISSUE THREAD

**Issue by elshobokshy** [ruflin/Elastica#1536]
I have a page that gets from an API source elasticsearch results. I want to be able, when I re-run the elasticsearch mapping, to show them a "This mapping is currently being rewritten" message or something similar.

I know in Elastica the only way to access such information is through the client class. http://elastica.io/api/latest/classes/Elastica.Client.html

```
$client = new \Elastica\Client(array('host' => SEARCH_ENGINE_ENDPOINT, 'port' => SEARCH_ENGINE_PORT));
```
But I can't find a way to get the information I'm looking for, that of knowing if a certain type is currently being filled.

Any idea if this is possible?

**Comment by p365labs** [comment]
What about using the getStats() at Index Level ?
maybe something like 

```php
$indexStats = $index->getStats();
$stats->get('docs', 'store', 'indexing', 'get');
```

or maybe have a look at all stats 
```php
$stats->get('_all');
```
are the stats returned what u want ?
Have a look at all documentations on [Indices Stats](https://www.elastic.co/guide/en/elasticsearch/reference/6.4/indices-stats.html) 

Another interesting endpoint is [Indices Recovery](https://www.elastic.co/guide/en/elasticsearch/reference/6.4/indices-recovery.html) which I'm not sure it could be a good fit for you, maybe have a look at that but at the moment is not yet implemented into  Elastica.

**Comment by elshobokshy** [comment]
Could you please add a small example on how to use this new class to get my desired result?

**Comment by p365labs** [comment]
@elshobokshy if u have a look into the PR merged #1537 u can view the test which uses the getRecovery() method. For the informations it returns please refer to the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/6.4/indices-recovery.html)
